libqtile/core/manager.py

```diff
@@ -595,7 +687,12 @@ def manage(self, w):
                 # Window may have been bound to a group in the hook.
                 if not c.group:
                     self.current_screen.group.add(c, focus=c.can_steal_focus())
+                parent = w.get_wm_transient_for()
+                if parent:
+                    c.window.net_wm_state_above = self.windows_map[parent].window.net_wm_state_above
+                    c.window.net_wm_state_below = self.windows_map[parent].window.net_wm_state_below
                 self.update_client_list()
+                self.change_layer(w.wid, force=True)
```

Comment:
This is the faulty line for the behavior I'm seeing when I restart under i3lock. What's the rationale behind it?

Reply:
Once a new window is added, it has to be moved to its place in the stack, otherwise we might violate the "pinned" state of other windows or even fall out of sync with the stacking order we report to X11.

Do you know whether i3lock has _NET_WM_STATE_ABOVE or anything similar set?

Reply:
It looks like it doesn't set anything but WM_NAME? And most importantly we don't manage its window (it's not listed in `qtile.windows_map` when running).

Reply:
If so, then that likely means that layering *any* window will lead to the unmanaged windows falling by the wayside? We could maybe work around that by using `StackMode.Below` exclusively (if nothing is ever raised, the unmnaged windows won't be buried), but that feels like a bit of a hack.

Reply:
After adding that most recent commit, the behaviour isn't perfect, but it is back to how the master-branch is currently behaving (which I just now noticed). So unless the final commit is too ugly, I would suggest that to be tackled in a separate PR, likely by someone more knowledgable than me.